public/app/plugins/panel/canvas/ConnectionSVG.tsx

```diff
@@ -0,0 +1,221 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useRef, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import { CanvasConnection } from 'app/features/canvas/element';
+import { ElementState } from 'app/features/canvas/runtime/element';
+import { Scene } from 'app/features/canvas/runtime/scene';
+
+type Props = {
+  setSVGRef: (anchorElement: SVGSVGElement) => void;
+  setLineRef: (anchorElement: SVGLineElement) => void;
+  scene: Scene;
+};
+
+interface ConnectionInfo {
+  source: ElementState;
+  target: ElementState;
+  info: CanvasConnection;
+}
+
+export const ConnectionSVG = ({ setSVGRef, setLineRef, scene }: Props) => {
+  const styles = useStyles2(getStyles);
+
+  const CONNECTION_LINE_ID = 'connectionLineId';
+  const CONNECTION_HEAD_ID = 'head';
+
+  const [selectedConnection, setSelectedConnection] = useState<CanvasConnection | undefined>(undefined);
+
+  // Need to use ref to ensure state is not stale in event handler
+  const selectedConnectionRef = useRef(selectedConnection);
+  useEffect(() => {
+    selectedConnectionRef.current = selectedConnection;
+  });
+
+  const [selectedConnectionSource, setSelectedConnectionSource] = useState<ElementState | undefined>(undefined);
+  const selectedConnectionSourceRef = useRef(selectedConnectionSource);
+  useEffect(() => {
+    selectedConnectionSourceRef.current = selectedConnectionSource;
+  });
+
+  const onKeyUp = (e: KeyboardEvent) => {
+    if (e.keyCode === 8) {
+      if (selectedConnectionRef.current && selectedConnectionSourceRef.current) {
+        selectedConnectionSourceRef.current.options.connections =
+          selectedConnectionSourceRef.current.options.connections?.filter(
+            (connection) => connection !== selectedConnectionRef.current
+          );
+        selectedConnectionSourceRef.current.onChange(selectedConnectionSourceRef.current.options);
+
+        setSelectedConnection(undefined);
+        setSelectedConnectionSource(undefined);
+      }
+    }
+
```

Comment:
```suggestion
} else {
      return;
    }
```

Reply:
Without this exit, when selecting an arrow and pressing any key other than backspace, I'm not able to unselect or delete.